st2common/st2common/bootstrap/runnersregistrar.py

```diff
@@ -134,8 +134,8 @@
                 'secret': True
             },
             'private_key': {
-                'description': ('Private key material to log in. Note: This needs to be actual '
-                                'private key data and NOT path.'),
+                'description': ('Private key material or path to the private key file on disk '
```

Comment:
In any case, I will add some docs on the limitations and security implications of this approach.

Then we can decide if we are fine with it or not. If we are not, I will open a new PR and at least cherry-pick some fixes, improvements and tests from this PR.